src/Grpc.StatusProto/RpcStatusExtensions.cs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Google.Rpc;
+using Grpc.Core;
+
+namespace Grpc.StatusProto;
+
+/// <summary>
+/// Extensions for <see cref="Google.Rpc.Status"/> to retrieve detailed error information.
+/// Based on ideas from:
+/// https://github.com/googleapis/gax-dotnet/blob/main/Google.Api.Gax.Grpc/RpcExceptionExtensions.cs
+/// </summary>
+public static class RpcStatusExtensions
+{
+    /// <summary>
+    /// Retrieves the error details of type <typeparamref name="T"/> from the <see cref="Google.Rpc.Status"/>
+    /// message.
+    /// </summary>
+    /// <typeparam name="T">The message type to decode from within the error details.</typeparam>
+    /// <param name="status">The RPC status to retrieve details from. Must not be null.</param>
+    /// <returns>The first error details of type <typeparamref name="T"/> found, or null if not present</returns>
+    public static T? GetStatusDetail<T>(this Google.Rpc.Status status) where T : class, IMessage<T>, new()
+    {
+        var expectedName = new T().Descriptor.FullName;
+        var any = status.Details.FirstOrDefault(a => Any.GetTypeName(a.TypeUrl) == expectedName);
+        if (any is null)
+        {
+            return null;
+        }
+        return any.Unpack<T>();
+    }
+
+    /// <summary>
```

Comment:
I know methods like this are present in GAX at the moment - but I'm not sure whether they're really worthwhile. It's easy enough to just call `GetStatusDetail<BadRequest>`.

I'd suggest sticking to just the generic method to start with (including some examples) and then adding more extension methods later if we need them.

Reply:
Looks like this was resolved?

Reply:
Yup. Don't think I can resolve it though.